src/kafka/producer.ts

```diff
@@ -1,60 +1,75 @@
-import { Kafka, Partitioners, logLevel } from  'kafkajs';
-import kafkaConfig from  './kafkaConfig.js';
-import alert from  'cli-alerts';
+import { Partitioners, Producer } from 'kafkajs';
+import kafkaConfig from './kafkaConfig.js';
+import alert from 'cli-alerts';
+import { OutputFormat } from '../formats/outputFormat.js';
+import createTopic from './createTopic.js';
 
-export async function kafkaProducer(producer: any, recordKey = null, record, encodedRecord = null, topic = 'datagen_test_topic') {
+export class KafkaProducer {
+    private producer: Producer;
+    private format: OutputFormat;
 
-    if (global.prefix) {
-        topic = `${global.prefix}_${topic}`;
+    static async create(format: OutputFormat): Promise<KafkaProducer> {
+        const kafka = await kafkaConfig();
+        const producer = kafka.producer({
+            createPartitioner: Partitioners.DefaultPartitioner
+        });
+
+        if (global.debug) {
+            console.log(`Connecting to Kafka producer...`);
```

Comment:
Should be "Connecting to Kafka broker...", right?

Reply:
yes, whoops